sdk/apimanagement/api-management-custom-widgets-tools/src/utils.ts

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/**
+ * Key for a search param, from which editor data will be loaded from.
+ */
+export const EDITOR_DATA_KEY = "editorData";
+/**
+ * Key for a post message object, it's used to propagate changes from editor to the DevPortal. Used to prevent interference with other applications.
+ */
+export const ON_CHANGE_MESSAGE_KEY = "customInputValueChangedMSAPIM";
+/**
+ * Key for a post message object, it's used to request and send secrets - token and user id, from the DevPortal. Used to prevent interference with other applications.
+ */
+export const ASK_FOR_SECRETS_MESSAGE_KEY = "askForSecretsMSAPIM";
+
+/**
+ * Base of a values obj
+ */
+export type TValuesCommon = Record<string, unknown>;
+/**
+ * All possible runtime environments
+ */
+export type TEnvironment = "development" | "publishing" | "runtime" | "error";
+
+/** Information about the widget instance received from the Dev Portal */
+export interface TPortalData {
+  /** web content's origin (URL) of your Dev Portal */
+  origin: string;
```

Comment:
originUrl?

Reply:
Changing PortalData keys would require refactor elsewhere and I don't see any added readability